tests/test_api_urls.py

```diff
@@ -66,31 +78,62 @@ def test_api_wrong_category(self):
+            api_response.content_type = 'application/json'
+            api_response.content = '{}'
+            api_response.headers = {
+                'etag': 'xxx',
+                'cache-control': 'private, max-age=60, s-maxage=60',
+                'content-type': 'appication/json',
+                'link': '<https://api.github.com/repositories/17839063/labels?page=2>; rel="next", <https://api.github.com/repositories/17839063/labels?page=2>; rel="last"'}  # nopep8
+            github_data.return_value = api_response
+            rv = self.app.get('/api/issues/labels', environ_base=headers)
+            self.assertEqual(rv.status_code, 200)
+            self.assertEqual(rv.content_type, 'application/json')
 
     def test_api_comments_link_header_auth(self):
         '''API access to comments greater than 30 returns pagination in Link
         header of the response.'''
-        query_string = {'callback': 'foo'}
-        # Force a JSONP callback response with `query_string` because it gives
-        # us the header properties we want to test.
-        rv = self.app.get('/api/issues/398/comments',
-                          query_string=query_string, environ_base=headers)
-        self.assertTrue = all(x in rv.data for x in ['Link', 'rel', 'next',
-                                                     'last', 'page'])
-        self.assertEqual(rv.status_code, 200)
-        self.assertEqual(rv.content_type, 'application/json')
-        # API access to comments for an issue with < 30 does not return link a
-        # header in the response (until GitHub changes it....?)
-        rv = self.app.get('/api/issues/4/comments', query_string=query_string,
-                          environ_base=headers)
-        self.assertTrue = not all(x in rv.data for x in ['Link', 'rel', 'next',
+        with patch('requests.get') as github_data:
+            # Create mock data. One api_response for each call.
+            api_response_398 = MagicMock(spec=Response)
+            api_response_398.status_code = 200
+            api_response_398.content_type = 'application/json'
+            api_response_398.content = '{"Link":[["https://api.github.com/repositories/17839063/issues/398/comments?callback=foo&page=2",{"rel":"next"}],["https://api.github.com/repositories/17839063/issues/398/comments?callback=foo&page=4",{"rel":"last"}]]}'  # nopep8
+            api_response_398.headers = {
+                'etag': 'xxx',
+                'cache-control': 'private, max-age=60, s-maxage=60',
+                'content-type': 'appication/json'}
+            api_response_4 = MagicMock(spec=Response)
+            api_response_4.status_code = 200
+            api_response_4.content_type = 'application/json'
+            api_response_4.content = '{}'
+            api_response_4.headers = {
+                'etag': 'xxx',
+                'cache-control': 'private, max-age=60, s-maxage=60',
+                'content-type': 'appication/json'}
```

Comment:
Hm, then we would have to add that to a lot of tests. Not only the ones we are changing here. And I only put that in, because, if not, our code would return an error. It doesn't make much difference to the tests themselves. Maybe open a new issue for that or just let it go?